### ISSUE TITLE
Dynamic load twmoji library

### ISSUE THREAD

**Issue by roderickhsiao** [joeattardi/picmo#123]
as we are using promise in some place when using `twmoji.parse`, wondering if we can centralized importing and dynamic load the library so it doesn't need to be included in main bundle if client are not using the twemoji style?

```js
private emitTwemoji(emoji: EmojiRecord): Promise<EmojiSelection> {
    return new Promise(resolve => {
      twemoji.parse(emoji.emoji, {
        ...this.options.twemojiOptions,
        callback: (icon, { base, size, ext }: any) => {
          const imageUrl = `${base}${size}/${icon}${ext}`;
          resolve({
            url: imageUrl,
            emoji: emoji.emoji,
            name: emoji.name
          });

          return imageUrl;
        }
      });
    });
```

**Comment by joeattardi** [comment]
Hey! Sorry for the long, long delay on this. Yikes!

I'm in the midst of a big refactor right now. One of the things I have working now is dynamic loading of the Twemoji and native emoji renderers.

So if you use `style: native`, it dynamically loads a chunk with the native emoji rendering code. The twemoji library is not loaded. If you use `style: twemoji`, it dynamically loads a chunk with the twemoji rendering code. Then the twemoji is loaded.

Seems to work well, though the bundle size savings of not using twemoji is fairly small. Still good though!

One idea I was thinking of, instead of using dynamic imports, is splitting the renderers up into separate npm packages. As one of the bigger consumers of emoji-button, I was wondering what you thought about that.

So one possible idea would be:

- a base package `@emoji-button/core`
- renderer packages: `@emoji-button/native`, `@emoji-button/twemoji`

Alternatively there could also be separate entry points but still under one npm package, then it would look something like:

```javascript
import EmojiButton from '@joeattardi/emoji-button';
import NativeEmoji from '@joeattardi/emoji-button/renderer/native';
import Twemoji from '@joeattardi/emoji-button/renderer/twemoji';

const picker = new EmojiButton({
  ...options,
renderer: new Twemoji(twemojiOptions)
});
```

I still haven't decided which direction to go, but regardless of which way I go I may be dropping the `@joeattardi`  scope from the package, it just feels amateurish to me. The reason I did it a while ago was so that I could also publish on the GitHub package registry, which at least at the time only worked with scoped packages.

I still do own the package name `emoji-button` because I initially published it under that name. I could switch back, that's one option, or I could do the scoped packages under `@emoji-button` as mentioned above.

Sorry for the long winded/brainstorming, curious to know as a consumer of the library which approach would be most desirable for you.

Thanks!

**Comment by roderickhsiao** [comment]
Thanks for the detail explanation! 

I personally think probably the alternative solution is more flexible as the library will be able to extend more potential emoji renderer library instead of having a switch condition within the library. although I do see the point that it might be easier for people to integrate just by passing the style flag.

**Comment by joeattardi** [comment]
With the new design, which is still a work in progress, Emoji Button is split up into several packages. 

- First you will need `@emoji-button/core` (the main picker package).
- Then you pick a renderer. Currently the supported renderers are native (`@emoji-button/renderer-native`) or twemoji (`@emoji-button/renderer-twemoji`).
- Finally, you pick the locale for the emoji and category data which will all be available in the `@emoji-button/emoji-data` package.

This is still a ways off but I'm getting there. Localized categories and names are working, sort of:

<img width="389" alt="image" src="https://user-images.githubusercontent.com/219285/120753194-d0cce100-c4d8-11eb-841b-e658ae43a407.png">

**Comment by joeattardi** [comment]
If you're curious to take a look, the latest code is on the `v5` branch, though I haven't pushed to GitHub in a while :)

**Comment by roderickhsiao** [comment]
Like it, nice work :))

**Comment by joeattardi** [comment]
This has been implemented and will be included in the 5.0 release which should be in beta soon!

The new version is quite differently structured and has a notion of renderers. If you want Twemoji you import the twemoji renderer separately, this means if you stick to the native renderer twemoji won't be included. 

Thanks for your patience on this!

**Comment by roderickhsiao** [comment]
Thanks for the hard work!! It's definitely not an easy task especially using your own time. Appreciated and can't wait for the new version :)

**Comment by joeattardi** [comment]
Thanks! I figured I should update some of these issues - I don't want people to think the project is dead!

**Comment by joeattardi** [comment]
This issue is resolved in `v5.0.0-beta.6`.

Twemoji is now a separate package, `@picmo/renderer-twemoji`.
